scripts/scene/RoomBuilder.gd

```diff
@@ -217,7 +223,21 @@ func draw_room(room) -> void:
 					build_tile_map.set_cell(building_layer, doorTile, tileset_id, Vector2(0, 1)) 
 				elif doorTile.y == max_y - 1:
 					build_tile_map.set_cell(building_layer, doorTile, tileset_id, Vector2(3, 2))
-
+					
+func save_base_tile_map_state() -> Dictionary:
+	var tile_data = {}
+	for x in range(base_tile_map.get_used_rect().position.x, base_tile_map.get_used_rect().position.x + base_tile_map.get_used_rect().size.x):
+		for y in range(base_tile_map.get_used_rect().position.y, base_tile_map.get_used_rect().position.y + base_tile_map.get_used_rect().size.y):
+			var coords = Vector2i(x, y)
+			var cell_atlas_data = base_tile_map.get_cell_atlas_coords(0, coords)
+			tile_data[coords] = cell_atlas_data
+	return tile_data
+
+func restore_base_tile_map_state() -> void:
+	base_tile_map.clear()
+	for coords in base_tile_map_data.keys():
+		var atlas_coords = base_tile_map_data[coords]
+		base_tile_map.set_cell(0, coords, 0, atlas_coords)
```

Comment:
@khatakslied sorry I remember you explaining this to me already but just so I get it 100% could you remind why we need to do this and why rebaking the navigation polygon didn't work?

Reply:
@conalli Manipulating the build layer didn't seem to affect the navigation mesh on the base layer, so I added navigation areas to tiles on both layers.

<img width="631" alt="Screenshot 2024-08-17 at 23 46 25" src="https://github.com/user-attachments/assets/9e43be34-7553-42e8-b5b8-fbc8a4af97d0">
<img width="596" alt="Screenshot 2024-08-17 at 23 46 18" src="https://github.com/user-attachments/assets/e8123304-5efd-4e4b-89b6-d86d0d20df86">

Now both tile layers interact over the same navigation layer. I had to remove the base tile whenever we put a room on top of it, because the base tile has navigation on the whole cell. Even over the walls. So I erase the cell underneath and restore it if we remove the room.

Reply:
Got it, that makes sense. Lets merge this and see what else we can work on